### PR TITLE
Plugin: Adds entry for cypress-mailhog email plugin

### PR DESCRIPTION
**Pull request by SMenigat**
[cypress-mailhog](https://github.com/SMenigat/cypress-mailhog) is a well maintained command library that allows interaction with the MailHog RestAPI. 

Since the library is already widely used I would be very happy to see it on the official plugin list. 

Please let me know if there are any open todos to get the plugin listed.

### CODE REVIEW

**Comment by SMenigat** [comment]
Is there any update on this?

File: src/data/plugins.json [hunk 1]
```diff
@@ -1206,6 +1206,13 @@
           "link": "https://github.com/mailslurp/cypress-mailslurp",
           "keywords": ["email", "mailslurp", "test", "commands"],
           "badge": "community"
+        },
+        {
+          "name": "cypress-mailhog",
+          "description": "A collection of usefull Cypress commands for testing Emails utilizing the MailHog RestAPI. Comes with TypeScript support.",
```
Comment (MikeMcC399) [inline src/data/plugins.json:1212]: Typo
usefull  > useful

Comment (SMenigat) [reply src/data/plugins.json]: Thanks, just fixed the typo. 

Note: the `Suggestion` Feature would make it possible for me to right away apply the suggested changes. This is very useful for smaller stuff like typos 😊 

image

Comment (MikeMcC399) [reply src/data/plugins.json]: Sure. I use that also, but didn't this time.

**Comment by mschile** [comment]
Hi @SMenigat 👋, thanks for submitting the plugin! Please check to see if it fulfills these requirements:

- [ ] 🚀 Works with the latest major version of Cypress
- [ ] 🛠 Plugin purpose is clearly documented (in a README or docs website)
- [ ] 📝 Well-written documentation - A great example ([link](https://github.com/Fredx87/cypress-keycloak-commands#cypress-keycloak-commands))
- [ ] 🔬 Tested using Cypress - tests using Cypress can act as both example usage _and_ test coverage
- [ ] 👷‍♀️ CI pipeline that's passing (CircleCI and [Github Actions](https://github.com/cypress-io/cypress-tutorial-build-todo/blob/master/.github/workflows) are both free for Open Source)

If you have reason to believe your plugin does not need to meet certain requirements, please feel free to provide justification.

**Comment by MikeMcC399** [comment]
According to https://github.com/SMenigat/cypress-mailhog/blob/master/package.json it is using a legacy version of Cypress

    "cypress": "7.6.0",

The current version of Cypress is `12.11.0`

**Comment by chrisbreiding** [comment]
@SMenigat I think the only thing holding this up is that the plugin is not currently tested against the latest version of Cypress. If you can address that, we can move forward with reviewing and merging this.

File: src/data/plugins.json (hunk 1 above)
Comment (MikeMcC399) [reply src/data/plugins.json]: @SMenigat 
You also have this typo in your repo README.

**Comment by SMenigat** [comment]
@chrisbreiding / @MikeMcC399 : I just upgraded the test-servers cypress version to v12.13.0 and released cypress-mailhog@2.1.0

Thanks for your patience. Please let me know if there is anything left to do to get this included into the plugin list 😊

**Comment by AtofStryker** [comment]
> @chrisbreiding / @MikeMcC399 : I just [upgraded the test-servers cypress version to v12.13.0](https://github.com/SMenigat/cypress-mailhog/pull/44) and released cypress-mailhog@2.1.0
> 
> Thanks for your patience. Please let me know if there is anything left to do to get this included into the plugin list 😊

Things I believe look good on our end and met the criteria in https://github.com/SMenigat/cypress-mailhog/pull/44. Would you also be able to update the Cypress version in the root directory from `7.6.0` to `12.13.0`?

**Review by AtofStryker — APPROVED** [review]

**Comment by SMenigat** [comment]
@AtofStryker done ✅ 

Thanks, I'm stoked to get this added 😄

**Comment by AtofStryker** [comment]
I am going to merge this in as I feel it now fits the criteria laid out in https://github.com/cypress-io/cypress-documentation/pull/5203#issuecomment-1536602583

> 
> * [x]  🚀 Works with the latest major version of Cypress
> * [x]  🛠 Plugin purpose is clearly documented (in a README or docs website)
> * [x]  📝 Well-written documentation - A great example ([link](https://github.com/Fredx87/cypress-keycloak-commands#cypress-keycloak-commands))
> * [x]  🔬 Tested using Cypress - tests using Cypress can act as both example usage _and_ test coverage
> * [x]  👷‍♀️ CI pipeline that's passing (CircleCI and [Github Actions](https://github.com/cypress-io/cypress-tutorial-build-todo/blob/master/.github/workflows) are both free for Open Source)

**Pull request merged**